subprojects/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/api/InvalidConfigurationResolutionIntegrationTest.groovy

```diff
@@ -64,12 +68,10 @@ class DeprecatedConfigurationsIntegrationTest extends AbstractIntegrationSpec {
         """
 
         when:
-        executer.expectDocumentedDeprecationWarning("The compile configuration has been deprecated for dependency declaration. This will fail with an error in Gradle 8.0. " +
-            "Please use the implementation configuration instead. " +
-            "Consult the upgrading guide for further information: https://docs.gradle.org/current/userguide/upgrading_version_5.html#dependencies_should_no_longer_be_declared_using_the_compile_and_runtime_configurations")
+        fails 'help'
 
         then:
-        succeeds 'help'
+        failure.hasErrorOutput("Dependencies can no longer be declared using the `compile` configuration.")
```

Comment:
Not sure how I feel about this error message. The `can no longer be` implies that they were previously able to declare against this configuration. If we're throwing an error that means we are setting `setCanBeDeclared(false)`. But the wording seems like it is more of a deprecation.

In other words, I don't think our error messages should need to specify previous behavior (`can no longer`) as opposed to declare a current behavior (`cannot be`). That would mean we need to remember each configuration which was previously declarable but is not anymore. 

I would expect the following behavior in each of the two situations:
`configuration.setCanBeDeclared(false)` -> throws exception with wording like: "Dependencies can not declared using the `compile` configuration."

`configuration.deprecateForDeclaration()` -> prints deprecation warning with wording like: "The compile configuration has been deprecated for dependency declaration"

What further confuses me is that this test is still testing the `depreacateFor` methods. We should be getting deprecation warnings here, not errors.

Reply:
Good points on the wording.  I'll improve the test methods and the error message.

I'm going to leave the project dependencies report for another PR.  Someone should look at that more holistically.